controllers/dnspolicy_controller_single_cluster_test.go

```diff
@@ -171,85 +180,97 @@ var _ = Describe("DNSPolicy Single Cluster", func() {
-										"SetIdentifier":    Equal("default"),
-										"RecordTTL":        Equal(externaldns.TTL(300)),
-										"ProviderSpecific": Equal(externaldns.ProviderSpecific{{Name: "geo-code", Value: "*"}}),
-									})),
-									PointTo(MatchFields(IgnoreExtras, Fields{
-										"DNSName":       Equal(TestHostOne),
-										"Targets":       ConsistOf("lb-" + lbHash + ".test.example.com"),
-										"RecordType":    Equal("CNAME"),
-										"SetIdentifier": Equal(""),
-										"RecordTTL":     Equal(externaldns.TTL(300)),
-									})),
-								),
-							}),
+
+				dnsRecord := &kuadrantdnsv1alpha1.DNSRecord{}
+				err = k8sClient.Get(ctx, client.ObjectKey{Name: recordName, Namespace: testNamespace}, dnsRecord)
+				g.Expect(err).NotTo(HaveOccurred())
+
+				wildcardDnsRecord := &kuadrantdnsv1alpha1.DNSRecord{}
+				err = k8sClient.Get(ctx, client.ObjectKey{Name: wildcardRecordName, Namespace: testNamespace}, wildcardDnsRecord)
+				g.Expect(err).NotTo(HaveOccurred())
+
+				g.Expect(*dnsRecord).To(
+					MatchFields(IgnoreExtras, Fields{
+						"ObjectMeta": HaveField("Name", recordName),
+						"Spec": MatchFields(IgnoreExtras, Fields{
+							"OwnerID":        Equal(&clusterID),
+							"ManagedZoneRef": HaveField("Name", "mz-example-com"),
+							"Endpoints": ConsistOf(
+								PointTo(MatchFields(IgnoreExtras, Fields{
+									"DNSName":       Equal(clusterID + "-" + gwHash + "." + "klb.test.example.com"),
+									"Targets":       ConsistOf(TestIPAddressOne, TestIPAddressTwo),
+									"RecordType":    Equal("A"),
+									"SetIdentifier": Equal(""),
+									"RecordTTL":     Equal(externaldns.TTL(60)),
+								})),
+								PointTo(MatchFields(IgnoreExtras, Fields{
+									"DNSName":          Equal("default." + "klb.test.example.com"),
+									"Targets":          ConsistOf(clusterID + "-" + gwHash + "." + "klb.test.example.com"),
+									"RecordType":       Equal("CNAME"),
+									"SetIdentifier":    Equal(clusterID + "-" + gwHash + "." + "klb.test.example.com"),
+									"RecordTTL":        Equal(externaldns.TTL(60)),
+									"ProviderSpecific": Equal(externaldns.ProviderSpecific{{Name: "weight", Value: "120"}}),
+								})),
+								PointTo(MatchFields(IgnoreExtras, Fields{
+									"DNSName":          Equal("klb.test.example.com"),
+									"Targets":          ConsistOf("default." + "klb.test.example.com"),
+									"RecordType":       Equal("CNAME"),
+									"SetIdentifier":    Equal("default"),
+									"RecordTTL":        Equal(externaldns.TTL(300)),
+									"ProviderSpecific": Equal(externaldns.ProviderSpecific{{Name: "geo-code", Value: "*"}}),
+								})),
+								PointTo(MatchFields(IgnoreExtras, Fields{
+									"DNSName":       Equal(TestHostOne),
+									"Targets":       ConsistOf("klb.test.example.com"),
+									"RecordType":    Equal("CNAME"),
+									"SetIdentifier": Equal(""),
+									"RecordTTL":     Equal(externaldns.TTL(300)),
+								})),
+							),
 						}),
```

Comment:
I would have probably have avoided re-factoring these tests as part of other work, but since you have half done it here, these would be better broken up like:

```
g.Expect(dnsRecord.Name).To(Equal(recordName))
g.Expect(dnsRecord.Spec.OwnerID).To(Equal(&clusterID))
g.Expect(dnsRecord.Spec.ManagedZoneRef).To(HaveField("Name", "mz-example-com"))
g.Expect(dnsRecord.Spec.Endpoints).To(ConsistOf(
	PointTo(MatchFields(IgnoreExtras, Fields{
		"DNSName":       Equal(clusterID + "-" + gwHash + "." + "klb.test.example.com"),
		"Targets":       ConsistOf(TestIPAddressOne, TestIPAddressTwo),
		"RecordType":    Equal("A"),
		"SetIdentifier": Equal(""),
		"RecordTTL":     Equal(externaldns.TTL(60)),
	})),
	PointTo(MatchFields(IgnoreExtras, Fields{
		"DNSName":          Equal("default." + "klb.test.example.com"),
		"Targets":          ConsistOf(clusterID + "-" + gwHash + "." + "klb.test.example.com"),
		"RecordType":       Equal("CNAME"),
		"SetIdentifier":    Equal(clusterID + "-" + gwHash + "." + "klb.test.example.com"),
		"RecordTTL":        Equal(externaldns.TTL(60)),
		"ProviderSpecific": Equal(externaldns.ProviderSpecific{{Name: "weight", Value: "120"}}),
	})),
	PointTo(MatchFields(IgnoreExtras, Fields{
		"DNSName":          Equal("klb.test.example.com"),
		"Targets":          ConsistOf("default." + "klb.test.example.com"),
		"RecordType":       Equal("CNAME"),
		"SetIdentifier":    Equal("default"),
		"RecordTTL":        Equal(externaldns.TTL(300)),
		"ProviderSpecific": Equal(externaldns.ProviderSpecific{{Name: "geo-code", Value: "*"}}),
	})),
	PointTo(MatchFields(IgnoreExtras, Fields{
		"DNSName":       Equal(TestHostOne),
		"Targets":       ConsistOf("klb.test.example.com"),
		"RecordType":    Equal("CNAME"),
		"SetIdentifier": Equal(""),
		"RecordTTL":     Equal(externaldns.TTL(300)),
	})),
))
```

or even maybe this for easier debugging when stuff starts failing:

```
g.Expect(dnsRecord.Name).To(Equal(recordName))
g.Expect(dnsRecord.Spec.OwnerID).To(Equal(&clusterID))
g.Expect(dnsRecord.Spec.ManagedZoneRef).To(HaveField("Name", "mz-example-com"))
g.Expect(dnsRecord.Spec.Endpoints).To(HaveLen(4))
g.Expect(dnsRecord.Spec.Endpoints).To(ContainElement(
	PointTo(MatchFields(IgnoreExtras, Fields{
		"DNSName":       Equal(clusterID + "-" + gwHash + "." + "klb.test.example.com"),
		"Targets":       ConsistOf(TestIPAddressOne, TestIPAddressTwo),
		"RecordType":    Equal("A"),
		"SetIdentifier": Equal(""),
		"RecordTTL":     Equal(externaldns.TTL(60)),
	})),
))
g.Expect(dnsRecord.Spec.Endpoints).To(ContainElement(
	PointTo(MatchFields(IgnoreExtras, Fields{
		"DNSName":          Equal("default." + "klb.test.example.com"),
		"Targets":          ConsistOf(clusterID + "-" + gwHash + "." + "klb.test.example.com"),
		"RecordType":       Equal("CNAME"),
		"SetIdentifier":    Equal(clusterID + "-" + gwHash + "." + "klb.test.example.com"),
		"RecordTTL":        Equal(externaldns.TTL(60)),
		"ProviderSpecific": Equal(externaldns.ProviderSpecific{{Name: "weight", Value: "120"}}),
	})),
))
g.Expect(dnsRecord.Spec.Endpoints).To(ContainElement(
	PointTo(MatchFields(IgnoreExtras, Fields{
		"DNSName":          Equal("klb.test.example.com"),
		"Targets":          ConsistOf("default." + "klb.test.example.com"),
		"RecordType":       Equal("CNAME"),
		"SetIdentifier":    Equal("default"),
		"RecordTTL":        Equal(externaldns.TTL(300)),
		"ProviderSpecific": Equal(externaldns.ProviderSpecific{{Name: "geo-code", Value: "*"}}),
	})),
))
g.Expect(dnsRecord.Spec.Endpoints).To(ContainElement(
	PointTo(MatchFields(IgnoreExtras, Fields{
		"DNSName":       Equal(TestHostOne),
		"Targets":       ConsistOf("klb.test.example.com"),
		"RecordType":    Equal("CNAME"),
		"SetIdentifier": Equal(""),
		"RecordTTL":     Equal(externaldns.TTL(300)),
	})),
))
```

Reply:
That all said, i would avoid making any more changes to these in this PR and would prefer that re-factor to be done against main.

Reply:
yes, this makes sense, I might do this in a separate PR though, keen to get this closed out and move on to the AWS Health checks.